packages/components/table/stories/table.stories.mdx

```diff
@@ -75,6 +74,95 @@ export const Template = ({ cellProps = () => ({}), ...args }) => {
   </Story>
 </Canvas>
 
+### CellConfig
+
+Tables are built by passing in the cellConfigs prop to the table. They represent
+the configuration for each cell within a row. In the story above, all rows are
+rendered by reading from the records directly.
+
+#### Example
+
+```jsx
+const CELL_CONFIGS = [
+  { dataKey: "name" },
+  { dataKey: "id" },
+  { dataKey: "myFacility.name" },
+  { dataKey: "level" },
+]
+
+const records = [
+  { name: "Jerry Seinfeld", id: 1, myFacility: { name: "Paris " }, level: 1 },
+  { name: "Cosmo Kramer", id: 2, myFacility: { name: "Paris " }, level: 7 },
+  { name: "Elaine Benes", id: 3, myFacility: { name: "Paris " }, level: 4 },
+  { name: "George Costanza", id: 4, myFacility: { name: "Paris " }, level: 0 },
+]
+```
+
+<table className="doc-table">
+  <thead>
+    <tr>
+      <th>Option</th>
+      <th>Description</th>
+    </tr>
+  </thead>
+  <tbody>
+    <tr>
+      <td>dataKey</td>
+      <td>
+        A unique identifier for the cell in a row. This can be propert of the
+        record being rendered or a nested property within the record being rendered.
+      </td>
+    </tr>
+    <tr>
+      <td>cellProps (optional)</td>
+      <td>
+        Props to be passed to the table cell when rendered. They can be any props
```

Comment:
Cell props takes a function similar to render so you have access to the rowData. If we make the recommended update below, I think we should do it for cellProps too for consistency